sunpy/io/_asdf.py

```diff
@@ -0,0 +1,105 @@
+from pathlib import Path
+from collections import OrderedDict
+
+import numpy as np
+
+import asdf
+
+from sunpy.io._header import FileHeader
+
+__all__ = ["write", "read", "get_header", "get_keys_name"]
+
+def write(fname, data, header, **kwargs):
+    """
+    Take ``(data, header)`` pairs and save it to an ASDF file.
+
+    Parameters
+    ----------
+    fname : `str`
+        File name, with extension.
+    data : `numpy.ndarray`
+        N-dimensional data array.
+    header : `dict`
+        A header dictionary.
+    """
+    dt = data.dtype
+    data = data.astype(dt)
+    asdf.AsdfFile({Path(fname).name
+: {"meta" : dict(header), "data" : data.astype(data.dtype)}}, lazy_load=False, **kwargs).write_to(fname)
```

Comment:
```suggestion
    asdf.AsdfFile({Path(fname).name: {"meta":dict(header), "data":data.astype(data.dtype)}}, lazy_load=False, **kwargs).write_to(fname)
```

I made a mistake, the dict whitespaces I always get incorrect.